### DISCUSSION TITLE
中文参考文献两个作者不能显示为“xx和xx”而是“xx et al.”或“xx and xx”

### DISCUSSION THREAD

**Discussion by dhblues** [tuna/thuthesis#969] [Q&A] [answered]
一个参考文献在正文中引用显示格式的问题，就是如果参考文献作者为两人（无论中文还是英文），在正文引用中，中文作者会显示为“”张三等“，而不是“张三和李四”，英文作者则会显示为“A et al.” 而不是“A and B”。如以下代码：
@article{xp2019,
	title={中国农村金融改革的回顾、反思与展望},
	author={谢平 and 刘海二},
	journal={南方金融},
	number={3},
	pages={3-11},
	year={2019},
	key={xie4ping2liu2hai3er2},
}
\begin{document}
\cite{xp2019}
\end{document}

显示为“谢平等（2019）”而不是“谢平和刘海二（2019）”，若将代码改为 \citet*{xp2019}，则显示为“谢平 and 刘海二（2019）”

**Comment by Harry-Chen** [comment, marked as answer]
GB/T 7714-2015 10.2.2 条规定：

![image](https://github.com/tuna/thuthesis/assets/2819727/b24d14d9-2131-4b85-933b-60d93c3e87ce)

**Reply by dhblues** [reply]
但现在格式要求两个作者的话，英文作者中间用“and”，中文作者中间用“和”，请问有什么办法实现吗？

**Reply by zepinglee** [reply]
> 但现在格式要求两个作者的话，英文作者中间用“and”，中文作者中间用“和”，请问有什么办法实现吗？

学校研究生《指南》的表述跟国标一致。当然这里的“多著者”表述可能有不同的理解，但是《指南》和国标都没有做出具体说明，也没给出示例，比如英文引注是使用“and”还是“&”，所以我倾向于两个作者也算作“多著者”的情况。

<img width="746" alt="Screenshot 2024-07-12 at 08 49 57" src="https://github.com/user-attachments/assets/8e16ba7e-8dd4-434f-ad30-9452ea61fd2b">

**Reply by zepinglee** [reply]
如果实在需要“甲和乙”、“A and B”的格式，可以进行以下修改，但是需要自己对格式负责。

如果使用 BibTeX，这里改为 `#3`。

https://github.com/tuna/thuthesis/blob/60661f7d140c192418cc9151b54a6c59a99fae23/thuthesis-author-year.bst#L68

https://github.com/tuna/thuthesis/blob/60661f7d140c192418cc9151b54a6c59a99fae23/thuthesis-author-year.bst#L199-L204

这一段改为以下代码。

```bst
FUNCTION {citation.and}
{ terms.in.macro
    { "{\biband}" }
    { entry.lang lang.zh =
        { "和" }
        { " and " }
      if$
    }
  if$
}
```

如果使用 `biblatex`，参考文档 [`texdoc biblatex-gb7714-2015`](http://mirrors.ctan.org/macros/latex/contrib/biblatex-contrib/biblatex-gb7714-2015/biblatex-gb7714-2015.pdf)。

**Reply by dhblues** [reply]
按照您指导修改后问题已解决，非常感谢。

**Reply by dhblues** [reply]
@zepinglee 您好，按照您指导的修改后，现在在正文中可以两个作者中间显示“和”，三个作者显示“等”了，但在参考文献部分又出了问题，无论中文还是英文作者，都是两个作者中间用 “and”来连接，三个和三个以上的作者的话，前面的是逗号，最后一个和倒数第二个之间是“and”连接。请问这个是要在bib文件里面做调整吗（我现在bib文件里面作者之间都用and连接的）？

**Reply by zepinglee** [reply]
你是不是改了 `bbl.and`？建议将修改后的 `.bst` 文件和 `.bib` 发到 <https://gist.github.com/> 方便我复现。

**Reply by dhblues** [reply]
好的，不过我登不上https://gist.github.com，请问还有什么其他的方式发给您？

**Reply by zepinglee** [reply]
> 好的，不过我登不上[https://gist.github.com，请问还有什么其他的方式发给您？](https://gist.github.com%EF%BC%8C%E8%AF%B7%E9%97%AE%E8%BF%98%E6%9C%89%E4%BB%80%E4%B9%88%E5%85%B6%E4%BB%96%E7%9A%84%E6%96%B9%E5%BC%8F%E5%8F%91%E7%BB%99%E6%82%A8%EF%BC%9F)

或者打包为 zip 格式直接发在本讨论中。

**Reply by dhblues** [reply]
好的，谢谢您。
[bst+bib.zip](https://github.com/user-attachments/files/17096057/bst%2Bbib.zip)

**Reply by zepinglee** [reply]
>  [bst+bib.zip](https://github.com/user-attachments/files/17096057/bst%2Bbib.zip)

我无法复现你的问题。见 [discussion\#969](https://github.com/tuna/thuthesis/tree/discussion%23969) 分支的内容。

https://github.com/tuna/thuthesis/blob/bff1e4b2e019328e9086d4f392bc2be6e6fb901d/thuthesis-example.bbl#L14-L50

<img width="241" alt="Screenshot 2024-09-27 at 12 34 11" src="https://github.com/user-attachments/assets/2c9009b1-29a9-4ee6-9fce-323388bbd5c7">

<img width="750" alt="Screenshot 2024-09-27 at 12 34 19" src="https://github.com/user-attachments/assets/92a9a963-48ac-49fd-b311-be22224e3581">

**Comment by youkaichao** [comment]
摸索出来的一点经验（感谢 @Harry-Chen 帮助🙏🏻）

如果只想改 `\citet` 的显示（同时不动reference里面的显示），可以改这里：

<img width="629" alt="image" src="https://github.com/user-attachments/assets/66d9efd8-e05b-4b38-898e-baff0fd16c9a" />
